src/reconcile/rec_write.c

```diff
@@ -1436,15 +1436,24 @@ __wt_rec_split_finish(WT_SESSION_IMPL *session, WT_RECONCILE *r)
  *     Move a saved WT_UPDATE list from the per-page cache to a specific block's list.
  */
 static int
-__rec_supd_move(WT_SESSION_IMPL *session, WT_MULTI *multi, WT_SAVE_UPD *supd, uint32_t n)
+__rec_supd_move(
+  WT_SESSION_IMPL *session, WT_MULTI *multi, WT_SAVE_UPD *supd, uint32_t n, bool *is_cleanp)
 {
     uint32_t i;
+    bool is_clean;
+
+    is_clean = *is_cleanp;
 
     WT_RET(__wt_calloc_def(session, n, &multi->supd));
 
-    for (i = 0; i < n; ++i)
+    for (i = 0; i < n; ++i) {
```

Comment:
Isn't reconciliation leave_dirty tracking something very similar to is_clean? I think its confusing that we now have two very similar bools being passed around here.

Reply:
leave dirty only indicates the state of the original page after the reconciliation. Here we need something that indicates the state of each splitted page after reconciliation. Some may be clean and some may be dirty. Probably I should revert this commit as well just to keep the minimum fix in this PR.